src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionAnnotatedTypeFactory.java

```diff
@@ -1,46 +1,95 @@
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import org.checkerframework.checker.dependencyinjection.qual.Bind;
+import org.checkerframework.checker.dependencyinjection.qual.BindAnnotatedWith;
+import org.checkerframework.checker.dependencyinjection.qual.BindBottom;
+import org.checkerframework.checker.dependencyinjection.utils.KnownBindingsValue;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.common.accumulation.AccumulationAnnotatedTypeFactory;
 import org.checkerframework.common.basetype.BaseTypeChecker;
 import org.checkerframework.common.reflection.ClassValAnnotatedTypeFactory;
+import org.checkerframework.common.reflection.ClassValChecker;
 import org.checkerframework.common.reflection.qual.ClassVal;
 import org.checkerframework.dataflow.cfg.ControlFlowGraph;
 import org.checkerframework.dataflow.cfg.block.Block;
 import org.checkerframework.dataflow.cfg.node.MethodAccessNode;
 import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
 import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.dataflow.expression.JavaExpression;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
 import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.framework.type.treeannotator.ListTreeAnnotator;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+import org.checkerframework.javacutil.AnnotationBuilder;
+import org.checkerframework.javacutil.AnnotationMirrorSet;
 import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.ElementUtils;
 import org.checkerframework.javacutil.TreeUtils;
 
-public class DependencyInjectionAnnotatedTypeFactory extends ClassValAnnotatedTypeFactory {
+public class DependencyInjectionAnnotatedTypeFactory extends AccumulationAnnotatedTypeFactory {
 
-  // TODO: Not a good idea to use this logger
-  Logger logger = Logger.getLogger(DependencyInjectionAnnotatedTypeFactory.class.getName());
+  ClassValAnnotatedTypeFactory classValATF = getTypeFactoryOfSubchecker(ClassValChecker.class);
+
+  private static HashMap<String, KnownBindingsValue> knownBindings = new HashMap<>();
 
   /** The {@code com.google.inject.AbstractModule.bind(Class<Baz> clazz)} method */
   private final List<ExecutableElement> bindMethods = new ArrayList<>(3);
 
   /** The {@code com.google.inject.binder.LinkedBindingBuilder.to(Class<? extends Baz> method */
   private final List<ExecutableElement> toMethods = new ArrayList<>(3);
 
+  /** The {@code com.google.inject.binder.LinkedBindingBuilder.toInstance() method */
+  private final List<ExecutableElement> toInstanceMethods = new ArrayList<>(1);
+
+  private final List<ExecutableElement> annotatedWithMethods = new ArrayList<>(2);
```

Comment:
Needs a (simple) Javadoc comment.

In a separate PR, we should enable [require-javadoc](https://github.com/plume-lib/require-javadoc)